### PR TITLE
Improve Elasticsearch logging by sending logs in batches

### PR DESCRIPTION
**Pull request by Jannis**
This makes `WithErrorDrain` obsolete.

### CODE REVIEW

**Comment by leodasvacas** [comment]
You're using `use` to import macros which was introduced in the latest stable. You can either use `#[macro_use]` to keep supporting the older Rust versions or remove the older versions from the test matrix.

File: core/src/log/elastic.rs [hunk 1]
```diff
@@ -125,28 +130,123 @@ pub struct ElasticDrainConfig {
 /// ```
 pub struct ElasticDrain {
     config: ElasticDrainConfig,
+    error_logger: Logger,
+    logs: Arc<RwLock<Vec<ElasticLog>>>,
```
Comment (leodasvacas) [inline core/src/log/elastic.rs]: We never call `.read` on this, so it can just be a mutex.

**Review by leoyvens — CHANGES_REQUESTED** [review]

File: core/src/log/elastic.rs (hunk 1 above)
Comment (Jannis) [reply core/src/log/elastic.rs]: Good point!

**Comment by Jannis** [comment]
@leodasvacas Addressed the macro importing issue and switched over to a `Mutex`. Care to take another look?

**Review by leoyvens — APPROVED** [review]

**Pull request merged**